integration-tests/gradle/src/main/resources/basic-kotlin-application-project/src/main/kotlin/org/acme/MyMainClass.kt

```diff
@@ -0,0 +1,13 @@
+package `basic-kotlin-application-project`.src.main.kotlin.org.acme
+
+import io.quarkus.runtime.Quarkus
+import io.quarkus.runtime.QuarkusApplication
+import io.quarkus.runtime.annotations.QuarkusMain
+
+@QuarkusMain(name = "my-main")
+class MyMainClass : QuarkusApplication {
```

Comment:
This all looks fine to me, in general, but I'm curious why this wouldn't have the super class in the check above.  or is that check intended for top level functions and this test just doesn't exercise that code path?

Reply:
Right. I could add another level of superclass in order to exercise the recursive lookup.